### PR TITLE
Add into DNS Resolver new functionality for DNS Security Policy as part of 2023-07-01-preview release

### PR DESCRIPTION
**Pull request by jamesvoongms**
## Description

The PR adds new functionality for DNS Security Policy including the four new resources:

- Dns Resolver Policy (aka DNS Security Policy)
- Dns Resolver Domain List
- Dns Security Rule
- Dns Resolver Policy Link (Vnet Link)

Swagger Spec PR: https://github.com/Azure/azure-rest-api-specs/pull/30707

## Mandatory Checklist

- Please choose the target release of Azure PowerShell. (⚠️**Target release** is a different concept from **API readiness**. Please click below links for details.)
  - [x] [General release](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Public preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Private preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Engineering build](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] No need for a release

- [x] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For SDK-based development mode, update `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`.
        * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. 
    * For autorest-based development mode, include the changelog in the PR description.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

**Comment by jamesvoongms** [comment]
Hi team, for the breaking change of systemData, that was removed from our class objects because it is now a part of the common inherited object now.

**Comment by jamesvoongms** [comment]
May I have some help with the build issue as well? The CI/CD seems to be trying to pull from [https://github.com/Azure/azure-rest-api-specs/blob/master/specification/dnsresolver/resource-manager/Microsoft.Network/preview/2023-07-01-preview/dnsresolver.json](https://github.com/Azure/azure-rest-api-specs/blob/master/specification/dnsresolver/resource-manager/Microsoft.Network/preview/2023-07-01-preview/dnsresolver.json'.)

rather than the /blob/{commit_hash}/

File: src/DnsResolver/DnsResolver.Autorest/README.md [hunk 1]
```diff
@@ -70,13 +67,14 @@ In this directory, run AutoRest:
 > see https://aka.ms/autorest
 
 ``` yaml
-commit: 37072829f795ce840b8085035e8adf4721602f34
+commit: 6286ea80b3ac7eecfc2f230d65b2034a656f10bf
 require:
-  - $(this-folder)/../../readme.azure.noprofile.md
+  - $(this-folder)/../../../tools/SwaggerCI/readme.azure.noprofile.md
```
Comment (Nickcandy) [inline src/DnsResolver/DnsResolver.Autorest/README.md:72]: This line should be the same with original

Comment (jamesvoongms) [reply src/DnsResolver/DnsResolver.Autorest/README.md]: Fixed, thank you

Comment (jamesvoongms) [reply src/DnsResolver/DnsResolver.Autorest/README.md]: Hi @Nickcandy , i am not seeing that the CI gives the error below due to the change here:

{BBC4C6AF-4129-4998-8728-3E03B65232F7}

Would you know what the reason for this is?

File: src/DnsResolver/DnsResolver.Autorest/README.md [hunk 2]
```diff
@@ -70,13 +67,14 @@ In this directory, run AutoRest:
 > see https://aka.ms/autorest
 
 ``` yaml
-commit: 37072829f795ce840b8085035e8adf4721602f34
+commit: 6286ea80b3ac7eecfc2f230d65b2034a656f10bf
 require:
-  - $(this-folder)/../../readme.azure.noprofile.md
+  - $(this-folder)/../../SwaggerCI/readme.azure.noprofile.md
```
Comment (Nickcandy) [inline src/DnsResolver/DnsResolver.Autorest/README.md:72]: This line should be the same with original

Comment (jamesvoongms) [reply src/DnsResolver/DnsResolver.Autorest/README.md]: Fixed thank you

**Comment by jamesvoongms** [comment]
Note: for the breaking changes, we have since removed the properties (eg. SystemDataCreatedByType) from our models because they are now being inherited from the base azure classes.

**Comment by vidai-msft** [comment]
> Note: for the breaking changes, we have since removed the properties (eg. SystemDataCreatedByType) from our models because they are now being inherited from the base azure classes.

@jamesvoongms Do you mean it won't cause breaking actually? Additionally, I see test failure in CI. Can you please take a look?

**Comment by jamesvoongms** [comment]
Hi @vidai-msft , the issues with the build should now be fixed and were due to changes that affected past recordings, could you take another look?

**Comment by vidai-msft** [comment]
@jamesvoongms If these are actually not breaking changes, can you please follow this doc https://github.com/Azure/azure-powershell/blob/main/documentation/Debugging-StaticAnalysis-Errors.md#breaking-changes to suppress?

**Comment by jamesvoongms** [comment]
@vidai-msft , I believe the latest changes should have fixed both the tests and the static analysis checks now, could you help run azp run?

**Comment by jamesvoongms** [comment]
To clarify the test changes were due to recording issues that I have now fixed but the static analysis checks are due to removal of properties in our owned objects because swagger team has made them available in a common interface.

**Comment by vidai-msft** [comment]
@jamesvoongms Only one test case failed. Could you please take another look?

**Comment by vidai-msft** [comment]
@jamesvoongms I noticed you chose the option `Public Preview`. May I confirm if you plan to include the module in the Az or a dedicated version with preview as a suffix like x.y.z-preview?

**Comment by jamesvoongms** [comment]
@vidai-msft , previously even for 2020 preview we just did regular versions without preview suffix. So I would say this will just be a straight version upgrade unless you would recommend the suffix. I have also addressed the last test failure, could you help re-run and review PR? Thank you

https://www.powershellgallery.com/packages/Az.DnsResolver/1.0.0

**Comment by vidai-msft** [comment]
> @vidai-msft , previously even for 2020 preview we just did regular versions without preview suffix. So I would say this will just be a straight version upgrade unless you would recommend the suffix. I have also addressed the last test failure, could you help re-run and review PR? Thank you
> 
> https://www.powershellgallery.com/packages/Az.DnsResolver/1.0.0

Thank you @jamesvoongms for the clarification. Then you should choose the `General release` in the `Mandatory Checklist` section. I have updated it.

**Review by vidai-msft — APPROVED** [review]
LGTM

**Comment by vidai-msft** [comment]
@jamesvoongms Please leave a message here as the change log.

**Comment by jamesvoongms** [comment]
Thanks @vidai-msft !

Changelog:

- Add 4 new DNS Resolver Policy (DNS Security Policy) resources into the commandlets
1. DNS Resolver Policy (DNS Security Policy)
2. DNS Security Rule
3. DNS Resolver Policy Link (DNS Security Policy Link)
4. DNS Resolver Domain List

**Pull request merged**